inference-engine/src/transformations/src/transformations/common_optimizations/division_to_zero_fp16_resolver.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include "transformations/common_optimizations/division_to_zero_fp16_resolver.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <openvino/opsets/opset8.hpp>
+#include "ngraph/rt_info.hpp"
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+
+NGRAPH_RTTI_DEFINITION(ov::pass::DivisionToZeroFP16Resolver, "DivisionToZeroFP16Resolver", 0);
+
+constexpr float normalized_fp16_min = 6.103515625e-05f;  // fp16 minimal normalized  value
```

Comment:
you seem to use minimal normal number: 0x0400.
I would propose to use hex representation for this number because preprocessor can behave differently for this number.
union {
float f;
uint32_t u;
} un;
un.u = 0x0400.
Why don't you use minimal subnormal numbers for fp16? Are they ok for division?

Reply:
I also thought to use Union, but came to a conclusion that it's impossible in our case. We cannot just place 0x0400 which corresponds to 16 bit into 32 bit float, if we do that we will get the wrong value. 

So I stored value in 32 bit floating point literal, as do many other developers, e.g. OpenCL
![image](https://user-images.githubusercontent.com/5703039/144474647-a1d39877-f634-4e80-85ee-2812dc9166d8.png)


About subnormal / denormal numbers.  As it was described in the article you shared denormal numbers are slow. Therefore i preferred normal value over denormal, this should not affect accuracy but can save some performance
https://randomascii.wordpress.com/2012/05/20/thats-not-normalthe-performance-of-odd-floats/